### ISSUE TITLE
Add options to choose min and max images per class 

### ISSUE THREAD

**Issue by tambetm** [NVIDIA/DIGITS#161]
Add options to choose min images and max images per class when adding new dataset. This would ignore folders (classes) that have less than or more than given number of images. These fields could be added to dataset creation form, below validation set creation options. This would in many cases allow using of original dataset folder without additional preprocessing.

**Comment by lukeyeager** [comment]
That's a good idea. I actually have this feature implemented already in the `parse_folder` tool:
https://github.com/NVIDIA/DIGITS/blob/v2.0.0-preview/tools/parse_folder.py#L494-L504

I just haven't exposed it through the web interface yet.

**Comment by tambetm** [comment]
Thanks @lukeyeager, I knew I had seen these options somewhere in Digits, but had forgotten where.

Still exposing them through web interface would be nice.

**Comment by gheinrich** [comment]
Should the min/max conditions apply to the training set only or should they also apply to the validation/test sets? Should we ignore min/max number of images per class when samples are specified through text files? Thanks.

**Comment by tambetm** [comment]
I think min/max should be applied before dividing it into train/validation/test set. For example if you set min to 10 and train/val/test split to 60/20/20, then each class will have at least 6 training samples, 2 validation samples and 2 test samples. In case it doesn't split evenly, I would round down validation and test numbers and use remaining for training, so all samples are still used.

I agree, that when samples are specified using text files, min/max constraints should be ignored.

  Tambet

**Comment by gheinrich** [comment]
Hi @tambetm, I have created a Pull Request https://github.com/NVIDIA/DIGITS/pull/191 for this. Would you like to check this addresses your request? I have added the option to select min/max number of samples for the training set only. I figured that this wasn't necessary if the user specifies the train/test sets directly.

**Comment by tambetm** [comment]
Sorry, I was on a vacation and now catching up with backlog. I'll give this a try ASAP.
